backend/capellacollab/core/logging.py

```diff
@@ -0,0 +1,155 @@
+        get_logger(request).info("request started")
+        response: Response = await call_next(request)
+        get_response_logger(request).info(
+            "request finished", status_code=response.status_code
+        )
+
+        return response
+
+
+class HealthcheckFilter(logging.Filter):
+    def filter(self, record: logging.LogRecord) -> bool:
+        return record.getMessage().find("/healthcheck") == -1
+
+
+class ReqLogAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        return (
+            "trace_id={trace_id} method={method} path={path} user={user} client={client} {msg_pref}{msg}".format(
+                trace_id=self.extra["trace_id"],
+                method=self.extra["method"],
+                path=self.extra["path"],
+                user=self.extra["user"],
+                client=self.extra["client"],
+                msg_pref="" if self.extra["chaining"] else 'message="',
+                msg=msg + ("" if self.extra["chaining"] else '"'),
+            ),
+            kwargs,
+        )
+
+
+class ErrorLogAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        error_code = kwargs.pop("error_code", self.extra["error_code"])
+        return (f'error_code={error_code} message="{msg}"', kwargs)
+
+
+class ResLogAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        status_code = kwargs.pop("status_code", self.extra["status_code"])
+        return (f'status_code={status_code} message="{msg}"', kwargs)
+
+
+def get_general_logger(name: str, log_leveL=LOGGING_LEVEL) -> logging.Logger:
+    logger: logging.Logger = logging.getLogger(name)
+    logger.addFilter(HealthcheckFilter())
+    logger.setLevel(log_leveL)
+
+    if not logger.handlers:
+        handler = logging.StreamHandler()
+        formatter = logging.Formatter(
+            "time=%(asctime)s level=%(levelname)s function=%(funcName)s %(message)s"
+        )
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+
+    return logger
+
+
+def get_general_log_args(
+    request: Request, chaining=False
+) -> t.Dict[str, t.Any]:
+    client = request.client
+    return {
+        "trace_id": request.state.trace_id,
+        "method": request.method,
+        "path": request.url.path,
+        "user": request.state.user_name,
+        "client": client[0] + ":" + str(client[1]) if client else "unknown",
+        "chaining": chaining,
+    }
+
+
+def get_logger(request: Request) -> logging.LoggerAdapter:
+    return ReqLogAdapter(
+        get_general_logger(request.url.path), get_general_log_args(request)
+    )
+
+
+def get_error_code_logger(request: Request) -> logging.LoggerAdapter:
+    req_logger = ReqLogAdapter(
+        get_general_logger(request.url.path, log_leveL=logging.ERROR),
+        get_general_log_args(request, chaining=True),
+    )
+    return ErrorLogAdapter(
+        req_logger, {"error_code": None, "request": request}
+    )
```

Comment:
I added this logger later, because you wanted a logger that provides the ability to specify an `error_code` that uniquely identifies the location of the generated log message (easy to search). However, since we now have only one logger where you can simply type `log.info(..., extra={"error_code": xy})`, this is no longer needed and I have removed it